cli/core/pkg/artifact/local.go

```diff
@@ -45,3 +45,27 @@ func (l *LocalArtifact) Fetch() ([]byte, error) {
 	}
 	return b, nil
 }
+
+// FetchTest reads test plugin artifact based on the local plugin artifact path
+// To fetch the test binary from the plugin we are using plugin binary path and creating test plugin path from it
+// If the plugin binary path is `artifacts/darwin/amd64/cli/cluster/v0.27.0-dev/tanzu-cluster-darwin_amd64`
+// then test plugin binary will be under `artifacts/darwin/amd64/cli/cluster/v0.27.0-dev/test/` directory
+func (l *LocalArtifact) FetchTest() ([]byte, error) {
+	// test plugin directory based on the plugin binary location
+	testPluginDir := filepath.Join(filepath.Dir(l.Path), "test")
+	dirEntries, err := os.ReadDir(testPluginDir)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to read test plugin directory")
+	}
+	if len(dirEntries) != 1 {
+		return nil, errors.Errorf("unable to find test plugin under test directory(%v). Found %v files under that directory", testPluginDir, len(dirEntries))
```

Comment:
nit: The error might be a little misleading, we could probably just say "Expected only 1 file under the directory, but found ..." . I'm saying this to cover the case where the test plugin is one of many files in the directory.